### PR TITLE
fix: `model.set_requires_gradient_sync(False)` should be called to turn off gradient synchronization in FSDP2.

### PR DESCRIPTION
**Pull request by EquationWalker**
In FSDP2, the model(`FSDPModule`) does not have `no_sync()` and instead calls  `model.set_requires_gradient_sync(False)`  to turn off gradient synchronization. See at [torch.distributed.fsdp.FSDPModule.set_requires_gradient_sync](https://docs.pytorch.org/docs/stable/distributed.fsdp.fully_shard.html#torch.distributed.fsdp.FSDPModule.set_requires_gradient_sync)

### CODE REVIEW

**Comment by S1ro1** [comment]
Can you fix quality @EquationWalker ?
